plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeUpdateBucketFunction.java

```diff
@@ -33,7 +33,7 @@ public DeltaLakeUpdateBucketFunction(int bucketCount)
     @Override
     public int getBucket(Page page, int position)
     {
-        Block row = page.getBlock(0).getObject(position, Block.class);
+        Block row = page.getBlock(0).getObject(position, Block.class).getObject(0, Block.class);
```

Comment:
Basically because rowId type structure changed. 
Previously it was something like:
```
(path VARCHAR, position BIGINT, partition VARCHAR)
```
I had to add all data columns to it so I could create preimage so my initial change was to just add them at the end
so the structure would look like :
```
(path VARCHAR, position BIGINT, partition VARCHAR, column1 TYPE1, column2 TYPE2, ...)
```
@findepi suggested that this can cause problems with naming conflicts.
So the idea was to change the type of rowId to avoid it, now the type looks like this:
```
(necessary_columns(path VARCHAR, position BIGINT, partition VARCHAR), data_columns(column1 TYPE1, column2 TYPE2, ...))
```
and to access the same value in this `getBucket` function the code needs to go deeper into the block structure. 
I don't like the way I did this, maybe there is a nicer way any suggestions?

Reply:
Thank you for the thorough explanation.

Optional: 

```
(path VARCHAR, position BIGINT, partition VARCHAR, data(column1 TYPE1, column2 TYPE2, ...))
```

Is this a viable alternative?

Reply:
I think it is ok, @findepi  WDYT?